pkg/kopia/cli/repository/storage/s3/s3_opts.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2024 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package s3
+
+import (
+	"github.com/kanisterio/safecli/command"
+
+	"github.com/kanisterio/kanister/pkg/kopia/cli"
+)
+
+var (
+	subcmdS3 = command.NewArgument("s3")
+)
+
+// optBucket creates a new bucket option with a given name.
+// If the name is empty, it returns ErrInvalidBucketName.
+func optBucket(name string) command.Applier {
+	if name == "" {
+		return command.NewErrorArgument(cli.ErrInvalidBucketName)
+	}
+	return command.NewOptionWithArgument("--bucket", name)
+}
+
+// optEndpoint creates a new endpoint option with a given endpoint.
+// If the endpoint is empty, the endpoint option is not set.
+func optEndpoint(endpoint string) command.Applier {
+	if endpoint == "" {
+		return command.NewNoopArgument()
+	}
+	return command.NewOptionWithArgument("--endpoint", endpoint)
+}
+
+// optPrefix creates a new prefix option with a given prefix.
+// If the prefix is empty, the prefix option is not set.
+func optPrefix(prefix string) command.Applier {
+	return command.NewOptionWithArgument("--prefix", prefix)
```

Comment:
Why aren`t we returning `command.NewNoopArgument()` over here if `prefix` is empty , the way we are doing over [here](https://github.com/kanisterio/kanister/pull/2659/files#diff-c8a5d40489b59da0012ce94134e6325909185be70772d76a8caba9dbf98ee7a8R55)

Reply:
In Kopia, `--prefix=` is a valid option, and our [current implementation](https://github.com/kanisterio/kanister/blob/f65bc3ca1ad1ea035a1ee793e7b011076d5877ac/pkg/kopia/command/storage/s3_args.go#L49) implements the same logic, we don't want to change it.